core/src/main/java/org/elasticsearch/common/lucene/search/function/FiltersFunctionScoreQuery.java

```diff
@@ -144,6 +144,9 @@ public Weight createWeight(IndexSearcher searcher, boolean needsScores) throws I
         for (int i = 0; i < filterFunctions.length; ++i) {
             subQueryNeedsScores |= filterFunctions[i].function.needsScores();
             filterWeights[i] = searcher.createNormalizedWeight(filterFunctions[i].filter, false);
+            //TODO: Not sure if we should add such a method because it does nothing for any score function except for query_function but
+            // I don't have a good idea where else to put it
```

Comment:
I left a comment in QueryFunction: I think we can get rid of this method by passing the IndexSearcher in the constructor